ghproxy/ghmetrics/ghpath.go

```diff
@@ -0,0 +1,424 @@
+package ghmetrics
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+func getSimplifiedPath(path string) string {
+	fragment := getFirstFragment(path)
+	var tree node
+	switch fragment {
+	case "repos":
+		// /repos/:owner/:repo
+		tree = node{
```

Comment:
Totally spit-balling here but:

```go
function l(fragment string, children ...node) node {
    return node{
        Node: l(fragment),
        children: children,
    }
}

function v(fragment string, children ...node) node {
    return node{
        Node: variable(fragment),
        children: children,
    }
}

// .... 
func getSimplifiedPath(path string) string {
	fragment := getFirstFragment(path)
	var tree node
	switch fragment {
	case "repos":
		// /repos/:owner/:repo
		tree = l("repos", v("owner", v("repo",
				l("branches", v("branch", l("protection",
						l("restrictions", l("users"), l("teams")),
						l("required_status_checks", l("contexts")),
						l("required_pull_request_reviews"),
						l("required_signatures"),
						l("enforce_admins")))),
		  // ...
		  )))
  }
}
```

Reply:
While it reduces the LOC, I feel like a lot of the readability gets lost.

Reply:
Agreed. Depends on what we want to achieve. Maybe a tree-specific storage format would be better

Reply:
I think readability is going to be more important. There's only a few situations I can envision where this code would be edited, and "GitHub changed their API" seems the most likely. That means that someone else might have to change this tree structure.

Reply:
I think we should try to condense it a little more -- having the string literals closer to each other helps to get the actual point across -- which are related to each other and how.

Reply:
I've went ahead and did the change. I got rid of the whole switch block and added a root-shadow element to have the whole structure in one tree.

Looking at it now, I feel the readability is better than before, while we also have the shorter initialization gained from Steve's proposal.

Let me know your thoughts.